src/variables/body/index.tsx

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { Variables } from '../index';
+
+import { ObjectInspector, ObjectLabel, ITheme } from 'react-inspector';
+
+import { ReactWidget } from '@jupyterlab/apputils';
+
+import React, { useState, useEffect } from 'react';
+import { ArrayExt } from '@phosphor/algorithm';
+
+export class Body extends ReactWidget {
+  constructor(model: Variables.IModel) {
+    super();
+    this.model = model;
+    this.addClass('jp-DebuggerVariables-body');
+  }
+
+  model: Variables.IModel;
+
+  render() {
+    return <VariableComponent model={this.model} />;
+  }
+}
+
+const defaultNodeRenderer = ({
+  depth,
+  name,
+  data,
+  isNonenumerable,
+  expanded,
+  theme
+}: {
+  depth: number;
+  name: string;
+  data: any;
+  isNonenumerable: boolean;
+  expanded: boolean;
+  theme?: string | Partial<ITheme>;
+}) => {
+  const label = data.name === '' || data.name == null ? name : data.name;
+  const value = data.value;
+
+  return depth === 0 ? (
+    <span>
+      <span>{label}</span>
+      <span>:</span>
+      <span>{data.length}</span>
+    </span>
+  ) : depth === 1 ? (
+    <span>
+      <span style={{ color: 'rgb(136, 19, 145)' }}>{label}</span>
+      <span>:</span>
+      <span>{value}</span>
+    </span>
+  ) : (
+    <ObjectLabel name={label} data={data} isNonenumerable={isNonenumerable} />
+  );
+};
+
+const VariableComponent = ({ model }: { model: Variables.IModel }) => {
+  const [data, setData] = useState(model.scopes || []);
+
+  useEffect(() => {
+    const updateScopes = (_: Variables.IModel, update: Variables.IScope[]) => {
+      if (ArrayExt.shallowEqual(data, update)) {
+        return;
+      }
+      setData(update);
+    };
+
+    model.scopesChanged.connect(updateScopes);
+
+    return () => {
+      model.scopesChanged.disconnect(updateScopes);
+    };
+  });
+
+  const List = () => (
+    <>
+      {data.map(scopes => (
+        <ObjectInspector
+          key={scopes.name}
+          data={scopes.variables}
+          name={scopes.name}
+          nodeRenderer={defaultNodeRenderer}
+          theme={theme}
+        />
+      ))}
+    </>
+  );
+
+  return <>{List()}</>;
+};
+
+const theme = {
+  BASE_FONT_FAMILY: 'var(--jp-code-font-family)',
+  BASE_FONT_SIZE: 'var(--jp-code-font-size)',
+  BASE_LINE_HEIGHT: 'var(--jp-code-line-height)',
+
+  BASE_BACKGROUND_COLOR: 'var(--jp-layout-color0)',
```

Comment:
Shouldn't it this color instead?

```suggestion
  BASE_BACKGROUND_COLOR: 'var(--jp-layout-color1)',
```

![image](https://user-images.githubusercontent.com/591645/67475784-28497080-f657-11e9-81de-8a4806b7adf2.png)